user_manual.md

```diff
@@ -1,146 +1,177 @@
-Specifically, the Eunoia language has the following builtin expressions:
+To enable that, the Eunoia language has the following builtin constants:
+
 - `Type`, denoting the kind of all types,
 - `->`, denoting the function type,
 - `_`, denoting (higher-order) function application,
 - `Bool`, denoting the Boolean type,
-- `true` and `false`, denoting values of type `Bool`.
+- `true` and `false`, denoting the two values of type `Bool`.
 
-> The core logic of the Ethos also uses several builtin types (e.g. `Proof` and `Quote`) which define the semantics of proof rules. These types are intentionally left hidden from the user. Details on these types can be found throughout this document. More details on the core logic of the Ethos can be found in a forthcoming publication.
+> __Note:__ The core logic of the Ethos also uses several builtin types (e.g. `Proof` and `Quote`) which define the semantics of proof rules. These types are intentionally to exposed to the Eunoia user. Details on then can be found throughout this document. More details on the core logic of the Ethos will be available in a forthcoming publication.
 
-In the following, we informally write BNF categories `<symbol>` to denote an SMT-LIB version 3.0 symbol, `<term>` to denote an SMT-LIB term and `<type>` to denote a term whose type is `Type`, `<typed-param>` has syntax `(<symbol> <type> <attr>*)` and binds `<symbol>` as a fresh parameter of the given type and attributes (if provided).
+In the following, we informally use the syntactic categories `<symbol>` to denote an SMT-LIB 3.0 symbol, `<term>` to denote an SMT-LIB term and `<type>` to denote a term whose type is `Type`. The syntactic category `<typed-param>` is defined, BNF-style, as `(<symbol> <type> <attr>*)`. It binds `<symbol>` as a fresh parameter of the given type and attributes (if provided).
 
 The following commands are supported for declaring and defining types and terms. The first set of commands are identical to those in SMT-LIB version 3.0:
+
 - `(declare-const <symbol> <type> <attr>*)` declares a constant named `<symbol>` whose type is `<type>`. Can be given an optional list of attributes (see [attributes](#attributes)).
-- `(declare-type <symbol> (<type>*))` declares a new type named `<symbol>` whose kind is given by the argument types and return type `Type`.
+<!-- Moved here because it is present in SMT-LIB 3 -->
+
+- `(declare-consts <lit-category> <type>)` declares the class of symbols denoted by the literal category to have the given type.
+
+- `(declare-type <symbol> (<type>*))` declares a new type constructor named `<symbol>` whose kind is `Type` if `<type>*` is empty. If `<type>*` is `<type_1> ... <type_n>`, then kind of `<symbol>` is `(-> <type_1> ... <type_n> Type)`.
+  <!-- Added-->
+  This is a derived command as it is a shorthand for
+  `(declare-const <symbol> Type)` if `<type>*` is empty, and for
+  `(declare-const <symbol> (-> <type>* Type))` otherwise.
+
+<!--CT Do we really need `define-type`? -->
```

Comment:
Not really, no.  I am keeping if for now.

Reply:
ok